pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -121,6 +130,38 @@ func (a *actuator) Reconcile(ctx context.Context, bastion *extensionsv1alpha1.Ba
 	return a.Client().Status().Patch(ctx, bastion, patch)
 }
 
+func getNetAndSubnet(ctx context.Context, a *actuator, cluster *extensions.Cluster) (vNet, subnet string, err error) {
+	var infrastructureStatus *gcp.InfrastructureStatus
+	worker := &extensionsv1alpha1.Worker{}
+
+	err = a.Client().Get(ctx, client.ObjectKey{Namespace: cluster.ObjectMeta.Name, Name: cluster.Shoot.Name}, worker)
+	if err != nil {
+		return "", "", err
+	}
+
+	if worker == nil || worker.Spec.InfrastructureProviderStatus == nil {
+		return "", "", errors.New("infrastructure provider status must be not empty for worker")
+	}
+
+	if infrastructureStatus, err = helper.InfrastructureStatusFromRaw(worker.Spec.InfrastructureProviderStatus); err != nil {
+		return "", "", err
+	}
+
+	if infrastructureStatus.Networks.VPC.Name == "" {
+		return "", "", errors.New("virtual network must be not empty for infrastructure provider status")
+	}
+
+	if len(infrastructureStatus.Networks.Subnets) != 1 {
```

Comment:
This is not a proper assumption here. There can be multiple subnets for different purposes e.g the internal loadbalancing network. You would need to filter for at least the subnet with purpose [nodes](https://github.com/gardener/gardener-extension-provider-gcp/blob/366f966f8084ba63e4abb52eda54cdacd3e14574/pkg/apis/gcp/types_infrastructure.go#L79).

Reply:
committed in [8194f18](https://github.com/gardener/gardener-extension-provider-gcp/pull/436/commits/8194f18025cd6300c4ca750c0872045595e96801)